Test for GridWorkflow creation with logging on

uchchwhash opened the pull request:
Add integration test to create a
`datacube.api.grid_workflow.GridWorkflow` object when logging is
enabled. In the current `develop` branch this creates a
`datacube.drivers.manager.DriverManager` object that has a member
logger object that cannot be serialized.

uchchwhash committed:
Test for GridWorkflow creation with logging on

Add integration test to create a
`datacube.api.grid_workflow.GridWorkflow` object when logging is
enabled. In the current `develop` branch this creates a
`datacube.drivers.manager.DriverManager` object that has a member
logger object that cannot be serialized.

uchchwhash committed:
Merge branch 'develop' into issue285

jeremyh commented on integration_tests/test_gridworkflow.py:
```diff
@@ -0,0 +1,10 @@
+    from logging import getLogger, StreamHandler
+
+    logger = getLogger(__name__)
+    logger.addHandler(StreamHandler())
```
Since this changes global scope, we should add a finally block that removes it again after doing the test 

(alternatively it could be done with a pytest fixture but that's probably overkill in this case)

jeremyh reviewed (commented):
Thanks Imam. I'm surprised how little code can reproduce this; it doesn't even need to run the workflow itself. I'd add a short line at the top to explain why it's testing this combination (what failed in the past), since it's not an obvious thing to test.

In the future we should add some tests that actually run the workflow, but that'll be a separate issue...

andrewdhicks committed:
Set minimum cloudpickle version for logger pickling

See https://github.com/cloudpipe/cloudpickle/pull/96

uchchwhash committed:
Cleanup after test

- Remove `StreamHandler` from logger after test is finished

uchchwhash committed:
Explain the purpose of the test

Add comment that mentions the interaction of `DriverManager` with
logging and pickling.
